src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1518,6 +1518,79 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             break;
         case NI_Sve_ConditionalSelect:
             return LowerHWIntrinsicCndSel(node);
+        case NI_Sve_SetFfr:
+        {
+            StoreFFRValue(node);
+            break;
+        }
+        case NI_Sve_GetFfrByte:
+        case NI_Sve_GetFfrInt16:
+        case NI_Sve_GetFfrInt32:
+        case NI_Sve_GetFfrInt64:
+        case NI_Sve_GetFfrSByte:
+        case NI_Sve_GetFfrUInt16:
+        case NI_Sve_GetFfrUInt32:
+        case NI_Sve_GetFfrUInt64:
+        {
+            LIR::Use use;
+            bool     foundUse = BlockRange().TryGetUse(node, &use);
+            if (foundUse)
+            {
+                // Remove GetFfr and replace the GetFfr's use with the lvaFfrRegister lclVar.
+                CreateFFRDefIfNeeded(node);
+
+                GenTree* lclVar = comp->gtNewLclvNode(comp->lvaFfrRegister, TYP_MASK);
+                BlockRange().InsertBefore(node, lclVar);
+                use.ReplaceWith(lclVar);
+                GenTree* next = node->gtNext;
+                BlockRange().Remove(node);
+                return next;
+            }
+            else
+            {
+                node->SetUnusedValue();
+            }
+
+            break;
+        }
+        case NI_Sve_LoadVectorFirstFaulting:
+        {
+            unsigned lclNum = comp->lvaFfrRegister;
+
+            LIR::Use use;
+            bool     foundUse = BlockRange().TryGetUse(node, &use);
+
+            if (foundUse)
+            {
+                if (m_ffrTrashed)
+                {
+                    // Consume the FFR register value from local variable to simulate "use" of FFR,
+                    // only if it was trashed. If it was not trashed, we do not have to reload the
+                    // contents of the FFR register.
+
+                    CreateFFRDefIfNeeded(node);
```

Comment:
I think this has to use `StoreFFRValue`, creating the local if it does not exist. As we discussed offline the block order may not necessarily reveal that there is a `SetFFR` in the function that executed before this node.

Reply:
> creating the local if it does not exist

Yes, I need to not have the check for `lclNum == BAD_VAR_NUM` inside `CreateFFRDefIfNeeded()`. `StoreFFRValue()` stores the value "after" the node, while `CreateFFRDefIfNeeded()` needs to store the value "before" the node, so will keep them separate.